tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,197 @@
+        .unwrap()
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    let vendor_registry_uri = vendor.registry.clone();
+    debug!(
+        "Found vendor container registry at uri: {}",
+        vendor_registry_uri
+    );
+
+    // Auto resolve the expected paths for the kit contents archive
+    let kit_path = publish_kit_args.kit_path.as_path();
+    let kit_name = kit_path.file_name().unwrap().to_string_lossy();
+
+    let kit_version = find_version(kit_path, &kit_name)?;
+
+    let mut platform_images = Vec::new();
+    for arch in ["aarch64", "x86_64"] {
+        let kit_filename = format!("{}-{}-{}.tar", &kit_name, &kit_version, arch);
+        let path = kit_path.join(&kit_filename);
+
+        let out = docker!(["load", format!("--input={}", path.display()).as_str(),]);
+        let out = String::from_utf8_lossy(&out);
+        let digest_expression =
+            Regex::new("(?<digest>sha256:[0-9a-f]{64})").context(error::RegexSnafu)?;
+        let caps = digest_expression.captures(&out).unwrap();
+        let digest = &caps["digest"];
+
+        let arch_specific_target_uri = format!(
+            "{}/{}:{}-{}",
+            vendor_registry_uri, kit_name, &kit_version, arch
+        );
+
+        docker!(["tag", digest, &arch_specific_target_uri,]);
+
+        info!(
+            "Pushing kit image for platform {} to {}",
+            arch, &arch_specific_target_uri
+        );
+        docker!(["push", &arch_specific_target_uri,]);
+
+        let docker_arch =
+            DockerArchitecture::try_from(arch).context(error::InvalidArchitectureSnafu)?;
+        platform_images.push((docker_arch, arch_specific_target_uri.clone()));
+    }
+
+    let target_uri = format!("{}/{}:{}", vendor_registry_uri, kit_name, kit_version);
+
+    let images: Vec<&str> = platform_images
+        .iter()
+        .map(|(_, image)| image.as_str())
+        .collect();
+
+    let mut manifest_create_args = vec!["manifest", "create"];
+    if vendor.insecure {
+        manifest_create_args.push("--insecure")
+    }
+    manifest_create_args.push(&target_uri);
+    manifest_create_args.extend_from_slice(&images);
+    docker!(manifest_create_args);
+
+    for (arch, image) in platform_images.iter() {
+        docker!([
+            "manifest",
+            "annotate",
+            format!("--arch={}", arch).as_str(),
+            &target_uri,
+            image,
+        ]);
+    }
+
+    info!("Pushing kit to {}", &target_uri);
+    docker!(["manifest", "push", &target_uri,]);
+
+    info!("Successfully published kit to {}", target_uri);
+
+    // Cleans up the transient images. These don't serve a purpose outside of publishing the kit
+    // and take up space.
+    info!("Cleaning up local images");
+    docker!(["manifest", "rm", &target_uri]);
+    for (_, image) in platform_images.iter() {
+        docker!(["rmi", image]);
+    }
```

Comment:
Does this mean we won't keep a local copy of an image after we push it to a kit? Seems like cleaning up the manifest is fine but we might need the image again right away to test the kit when building variants.

Reply:
alright I'll keep the images around